de.dlr.sc.virsat.model.edit.test/src/de/dlr/sc/virsat/model/concept/lists/TypeSafeComposedPropertyBeanListTest.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2020 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.concept.lists;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.concept.list.TypeSafeComposedPropertyBeanList;
+import de.dlr.sc.virsat.model.concept.types.category.IBeanCategoryAssignment;
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyComposed;
+import de.dlr.sc.virsat.model.dvlm.categories.CategoriesFactory;
+import de.dlr.sc.virsat.model.dvlm.categories.Category;
+import de.dlr.sc.virsat.model.dvlm.categories.CategoryAssignment;
+import de.dlr.sc.virsat.model.dvlm.categories.propertydefinitions.ComposedProperty;
+import de.dlr.sc.virsat.model.dvlm.categories.propertydefinitions.PropertydefinitionsFactory;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.ArrayInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.ComposedPropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.PropertyinstancesFactory;
+import de.dlr.sc.virsat.model.dvlm.categories.util.CategoryAssignmentHelper;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.concepts.ConceptsFactory;
+
+public class TypeSafeComposedPropertyBeanListTest {
+
+	private ComposedProperty testProperty;
+	private BeanPropertyComposed<IBeanCategoryAssignment> bean;
+	private ComposedPropertyInstance cpi;
+	private TypeSafeComposedPropertyBeanList<IBeanCategoryAssignment> beanList;
+	private static final String PROPERTY_NAME = "testProperty";
+
+	@Before
+	public void setUp() throws Exception {
+		Concept testConcept = ConceptsFactory.eINSTANCE.createConcept();
+		Category testCategory = CategoriesFactory.eINSTANCE.createCategory();
+		testProperty = PropertydefinitionsFactory.eINSTANCE.createComposedProperty();
+		testProperty.setName(PROPERTY_NAME);
+		testProperty.setArrayModifier(PropertydefinitionsFactory.eINSTANCE.createDynamicArrayModifier());
+		
+		testConcept.getCategories().add(testCategory);
+		testCategory.getProperties().add(testProperty);
+		
+		CategoryAssignment testCA = CategoriesFactory.eINSTANCE.createCategoryAssignment();
```

Comment:
The section of Instantiating the CA could maybe simplified by usign the CategoryInstantiator.